### PR TITLE
Collections tree API endpoint results should always return :children key

### LINKED ISSUE #14482: UI should update when a collection changes parent
**Describe the bug**
There are two parts to this issue but both feel related.

## 1. The first part is a follow up from this PR (#14367), and is partially documented in the original issue #14280.

When a collection is moved to a new parent, that parent should be immediately open. #14367 is making sure that child collections gets moved, but the parent collection stays collapsed, making this UX really confusing. Note that the title ("First collection") indicates that the collection is currently open. However, you can't see it in the sidebar.
image

**Logs**
N/A

**To Reproduce**
- Follow the steps from #14280

**Expected behavior**
New parent collection should be immediately open, clearly showing its children (with the focus on a collection that was moved inside of it).

**Screenshots**
Please note the thin dark blue bar above "New collection". It disappears on a click wherever in the page. That leads us to the second part of this issue.
image

---

## 2. When we move **nested** child collection from an existing parent to the root collection, UI is showing that collection in both places. Clicking on it shows updates in both places as well. That disappears on a page refresh.

**Steps to reproduce:
- Make sure you haveone parent collection with at least two nested children collections
- Move the child to the root (Our analytics)
- Note that you can see it in both places
- Refresh the page and it's going to be clearly visible in the root only

**Screenshots**

**Information about your Metabase Installation:**
- Your browser and the version: Brave, Chrome, Firefox (doesn't seem to be related)
- Your databases: H2, local dev
- Metabase version: `v0.38.1-SNAPSHOT`

**Severity**
Both issues are really confusing. Not blocking anything, so I'd say it's a P2?

### PR DESCRIPTION
**Pull request by camsaul**
So it turns out the `metabase/entities/collections/FETCH_LIST` Redux action was getting correctly fired -- the Collections tree **was** being fetched by hitting the API endpoint after moving a Collection -- but because the `GET /api/collection/tree` endpoint didn't include a `:children` key for Collections that didn't have any Children, the state wasn't getting updated correctly when we moved the last child Collection out of a parent Collection. **`FETCH_LIST` does not replace the objects in the Redux store, but merges whatever's fetched with whatever is already in the store**. 

So by tweaking the API endpoint to always return `:children` (even when empty) the state now gets updated correctly. I'd imagine the behavior of `FETCH_LIST` that merges objects rather than replacing the list entirely is responsible for all sorts of other bugs in our application, but we can figure that out later 💀 

Fixes #14482

### CODE REVIEW

File: frontend/src/metabase/collections/components/CollectionEditMenu.jsx [hunk 1]
```diff
@@ -4,12 +4,7 @@ import { t } from "ttag";
 import EntityMenu from "metabase/components/EntityMenu";
 import { ANALYTICS_CONTEXT } from "metabase/collections/constants";
 
-export default function CollectionEditMenu({
-  isRoot,
-  isAdmin,
-  collectionId,
-  tooltip,
-}) {
+export default function CollectionEditMenu({ isRoot, collectionId, tooltip }) {
```
Comment (camsaul) [inline frontend/src/metabase/collections/components/CollectionEditMenu.jsx:7]: `isAdmin` is unused, Emacs was giving me a squiggly line

**Comment by nemanjaglumac** [comment]
Everything seems to be working for me locally.
I'll just need to update Cypress - something really weird is going on with (re)render, or the UI update. Prior to this fix, Cypress was giving me false positives without a certain "hack", and now (after the fix), it's complaining that it cannot click on an element detached from DOM. 🤯

I'm preparing test update that would accommodate both states (pre and post fix).

**Comment by nemanjaglumac** [comment]
@camsaul - I have updated the test and it now works as expected pre and post fix. Which means it will catch potential regression in the future. I was testing it locally for both scenarios.

[Cypress failure in this latest run](https://app.circleci.com/pipelines/github/metabase/metabase/12527/workflows/d9f07339-31bf-4ff2-b7b2-c01ff386fca6/jobs/469833) is that weird EE_520 regression that is not visible on `master` nor on `release-x.38.x` branch but is failing on different branches.

One more thing to note:
- This covers the scenario where nested collection is moved into the ROOT collection and is immediately "open" (you can see its children), but
- If we move any collection into a new parent, that new parent is NOT open by default and you can't see moved collection immediately (this is something @kdoh said should be the default behavior) - I will open a new issue for this

File: src/metabase/models/collection.clj [hunk 2]
```diff
@@ -1032,9 +1032,8 @@
...
        ([m]
-        (let [vs (for [v (vals m)]
-                   (cond-> v
-                     (:children v) (update :children ->tree)))]
-          (sort-by (fn [{coll-name :name, coll-id :id}]
-                     [((fnil u/lower-case-en "") coll-name) coll-id]) vs))))
+        (->> (vals m)
+             (map #(update % :children ->tree))
+             (sort-by (fn [{coll-name :name, coll-id :id}]
+                        [((fnil u/lower-case-en "") coll-name) coll-id])))))
```
Comment (paulrosenzweig) [inline src/metabase/models/collection.clj:1038]: Separate from the sketchiness of merging the entities, I do like the consistency of always including it. It's nice to be able to rely on having an array.

**Review by paulrosenzweig — APPROVED** [review]
LGTM

**Comment by mazameli** [comment]
Status of this is that it's blocked on test fixing?

**Comment by nemanjaglumac** [comment]
> Status of this is that it's blocked on test fixing?

@mazameli No. Test is fixed. Cypress (dashboard) is keeping these queued since yesterday, and we have ee520 failure (which isn't related to this). We can force-merge it, afaic.
image

**Pull request merged**